parsl/monitoring/db_manager.py

```diff
@@ -463,17 +463,28 @@ def start(self,
 
                 """
                 BLOCK_INFO messages
-
                 """
                 block_info_messages = self._get_messages_in_batch(self.pending_block_queue)
+                # Creation of dictionary to store previous block messages to avoid duplicate messages
+                UBM = {}  # type: Dict[Any, Any]
```

Comment:
having this initialisation of `UBM` here means that it is reset to empty on every iteration of the processing loop which begins around line 340. That means that this code will forget everything on each iteration of that loop - so it will forget that it has already added, for example, block 0 in an earlier iteration and add it again.

Probably the right thing to do is move this `UBM = {}` line to before the while loop starting around line 343